flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -134,6 +136,7 @@ class IcebergFilesCommitter extends AbstractStreamOperator<Void>
   public void initializeState(StateInitializationContext context) throws Exception {
     super.initializeState(context);
     this.flinkJobId = getContainingTask().getEnvironment().getJobID().toString();
+    this.operatorUniqueId = getRuntimeContext().getOperatorUniqueID();
```

Comment:
This can be a problem. it gets the `jobVertexId` (a UUID like string). I verified it by setting the `uidPrefix=iceberg-sink` in `TestFlinkIcebergSink`.
```
    FlinkSink.forRow(dataStream, SimpleDataUtil.FLINK_SCHEMA)
        ...
        .uidPrefix("iceberg-sink")
        .append();
````

Then I can see the `operatorUniqueId` value as jobVertexId of `0e4eaa3db14ae3b6752b25172fd05c50`. It can change when the job graph changed (e.g. adding a stateless map operator).  Then we won't be able to match the operator id properly. 

Ideally, we want to use the [operator uid from Flink](https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/ops/upgrading/#datastream-api), which can be set by users. See the code from `FlinkSink`. 
```
      if (uidPrefix != null) {
        committerStream = committerStream.uid(uidPrefix + "-committer");
      }
```

if uid is not provided by application, Flink automatically generates a UUID by default. Setting uid for stateful operator (like Iceberg committer) is a [best practice](https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/ops/production_ready/#set-uuids-for-all-operators). Flink uses uid to match state description. if uid changed, old state won't be matched and restored. uid works perfectly for this purpose. But I am not sure what is the best way to retrieve the `uid` in the `initializeState` method. 

I can see the operator name from debugger, which is also set by `FlinkSink` based on `uidPrefix`. It will be unique as `uidPrefix` should be unique. Nonetheless, it is still not the ideal candidate/replacement for `uid` though.
<img width="630" alt="image" src="https://user-images.githubusercontent.com/1545663/210861950-3bdf1442-63d3-42ad-b6d5-88d5bd4c3bd1.png">

Reply:
I did more experiment. if the `uidPrefix` is set in `FlinkSink`, then it seems that `getOperatorUniqueID`  does return a stable ID even if the job DAG changed.

if `uidPrefix` is not set in `FlinkSink`, DAG change leads to unstable/different values from `getOperatorUniqueID`. This is not really a problem, because we should always set `uidPrefix` for `FlinkSink` as best practice for stateful committer operator.

In summary, using `getOperatorUniqueID` seems acceptable to me.

Reply:
cc @hililiwei @openinx and see if they have other suggestions.

Reply:
`operatorUniqueId` is deterministaclly generated based on the uid of the operator which is used already by Flink to assign the state to this operator. This is competely correct.